server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMailboxesMethodTest.java

```diff
@@ -2087,29 +2090,98 @@ public void setMailboxesShouldReturnNotUpdatedErrorWhenMovingMailboxTriggersName
-                  "          \"parentId\" : \"" + mailboxRootAId.serialize() + "\"" +
-                  "        }" +
-                  "      }" +
-                  "    }," +
-                  "    \"#0\"" +
-                  "  ]" +
-                  "]";
-
-      given()
-          .header("Authorization", accessToken.serialize())
-          .body(requestBody)
-      .when()
-          .post("/jmap")
-      .then()
-          .statusCode(200)
-          .body(NAME, equalTo("mailboxesSet"))
-          .body(ARGUMENTS + ".notUpdated", hasEntry(equalTo(mailboxChildToMoveCId.serialize()), Matchers.allOf(
-                  hasEntry(equalTo("type"), equalTo("invalidArguments")),
-                  hasEntry(equalTo("description"), equalTo("Cannot rename a mailbox to an already existing mailbox.")))));
-    }
-}
+            "[" +
+                "  [ \"setMailboxes\"," +
+                "    {" +
+                "      \"update\": {" +
+                "        \"" + mailboxChildToMoveCId.serialize() + "\" : {" +
+                "          \"parentId\" : \"" + mailboxRootAId.serialize() + "\"" +
+                "        }" +
+                "      }" +
+                "    }," +
+                "    \"#0\"" +
+                "  ]" +
+                "]";
+
+        given()
+            .header("Authorization", accessToken.serialize())
+            .body(requestBody)
+            .when()
+            .post("/jmap")
+            .then()
+            .statusCode(200)
+            .body(NAME, equalTo("mailboxesSet"))
+            .body(ARGUMENTS + ".notUpdated", hasEntry(equalTo(mailboxChildToMoveCId.serialize()), Matchers.allOf(
+                hasEntry(equalTo("type"), equalTo("invalidArguments")),
+                hasEntry(equalTo("description"), equalTo("Cannot rename a mailbox to an already existing mailbox.")))));
+    }
+
+    @Test
+    public void delegateMailboxShouldUpdateParentMailboxAsLookupRight() throws Exception {
+        String shared = "shared";
+        String sharedSubMailbox = "shared.subMailbox";
+        String user = "sharee";
+        mailboxProbe.createMailbox(MailboxConstants.USER_NAMESPACE, username, shared);
+        MailboxId mailboxId = mailboxProbe.createMailbox(MailboxConstants.USER_NAMESPACE, username, sharedSubMailbox);
+
+        with()
+            .header("Authorization", accessToken.serialize())
+            .body("[" +
+                "  [ \"setMailboxes\"," +
+                "    {" +
+                "      \"update\": {" +
+                "        \"" + mailboxId.serialize() + "\" : {" +
+                "          \"sharedWith\" : {\"" + user + "\": [\"a\", \"w\"]}" +
+                "        }" +
+                "      }" +
+                "    }," +
+                "    \"#0\"" +
+                "  ]" +
+                "]")
+            .post("/jmap");
+
+        awaitMethod.run();
+
+        MailboxACL mailboxACL = aclProbe.retrieveRights(MailboxPath.forUser(username, shared));
```

Comment:
Could we do that with a JMAP GetMailboxes in order to stay on the protocol level?